pkg/engines/jenkins/controller.go

```diff
@@ -0,0 +1,456 @@
+		}
+	}
+
+	var err error
+	if originalLighthouseJob.Status.State != lighthouseJob.Status.State {
+		c.log.WithFields(jobutil.LighthouseJobFields(&lighthouseJob)).
+			WithField("from", originalLighthouseJob.Status.State).
+			WithField("to", lighthouseJob.Status.State).Info("Transitioning states.")
+		// make sure to set an activity record this job state update
+		c.addActivity(&lighthouseJob)
+		_, err = c.lighthouseClient.UpdateStatus(&lighthouseJob)
+	}
+	return err
+}
+
+func (c *Controller) syncAbortedJob(lighthouseJob v1alpha1.LighthouseJob, jenkinsBuilds map[string]Build) error {
+	if lighthouseJob.Status.State != v1alpha1.AbortedState || lighthouseJob.Complete() {
+		return nil
+	}
+
+	if build, exists := jenkinsBuilds[lighthouseJob.Name]; exists {
+		if err := c.jenkinsClient.Abort(getJobName(&lighthouseJob.Spec), &build); err != nil {
+			return fmt.Errorf("failed to abort Jenkins build: %v", err)
+		}
+	}
+
+	lighthouseJob.SetComplete()
+	c.addActivity(&lighthouseJob)
+
+	_, err := c.lighthouseClient.UpdateStatus(&lighthouseJob)
+	return err
+}
+
+func (c *Controller) syncTriggeredJob(lighthouseJob v1alpha1.LighthouseJob, jenkinsBuilds map[string]Build) error {
+	// Record last known state so we can patch
+	originalLighthouseJob := lighthouseJob.DeepCopy()
+
+	if _, exists := jenkinsBuilds[lighthouseJob.ObjectMeta.Name]; !exists {
+		// Do not start more jobs than specified.
+		if !c.canExecuteConcurrently(&lighthouseJob) {
+			return nil
+		}
+		buildID, err := c.getBuildID()
+		if err != nil {
+			return fmt.Errorf("error getting build ID: %v", err)
+		}
+		// Start the Jenkins job.
+		if err := c.jenkinsClient.Build(&lighthouseJob, buildID); err != nil {
+			c.log.WithError(err).WithFields(jobutil.LighthouseJobFields(&lighthouseJob)).Warn("Cannot start Jenkins build")
+			lighthouseJob.Status.State = v1alpha1.ErrorState
+			lighthouseJob.Status.Description = "Error starting Jenkins job."
+			lighthouseJob.SetComplete()
+		} else {
+			lighthouseJob.Status.State = v1alpha1.PendingState
+			lighthouseJob.Status.Description = "Jenkins job enqueued."
+			lighthouseJob.Status.StartTime = metav1.Now()
+		}
+	} else {
+		// If a Jenkins build already exists for this job, advance the LighthouseJob to Pending and
+		// it should be handled by syncPendingJob in the next sync.
+		lighthouseJob.Status.State = v1alpha1.PendingState
+		lighthouseJob.Status.Description = "Jenkins job enqueued."
+	}
+
+	var err error
+	if originalLighthouseJob.Status.State != lighthouseJob.Status.State {
+		c.log.WithFields(jobutil.LighthouseJobFields(&lighthouseJob)).
+			WithField("from", originalLighthouseJob.Status.State).
+			WithField("to", lighthouseJob.Status.State).Info("Transitioning states.")
+		// make sure to set an activity record this job state update
+		c.addActivity(&lighthouseJob)
+		_, err = c.lighthouseClient.UpdateStatus(&lighthouseJob)
+	}
+
+	return err
+}
+
+func (c *Controller) addActivity(lighthouseJob *v1alpha1.LighthouseJob) {
```

Comment:
@abayer - that seems to cut it for now. 

That said, is it not a bit odd that the _ActivityRecord_ just mirrors what's anyways already stored on the LighthouseJob?